### PR TITLE
Fix deprecation warnings in the test environment

### PR DESCRIPTION
**Pull request by goreck888**
Fix all warnings caused by ruby and rails upgrade
by running rspec

### CODE REVIEW

File: spec/features/tours/profile_spec.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@
 RSpec.feature "Profile tour" do
   include OmniauthHelper
 
-  LATER_COOKIE_NAME = "tours-marketplace-services-show-service_about_intro"
+  let(:later_cookie_name) { "tours-marketplace-services-show-service_about_intro" }
```
Comment (jswk) [inline spec/features/tours/profile_spec.rb:8]: I would remove it altogether, as it isn't used in this test anyway.

**Review by jswk — CHANGES_REQUESTED** [review]
Not exactly scope of the ticket, but I've noticed that we also have another shared examples: `spec/models/shared_examples_spec.rb`.

Maybe it could be removed.

**Comment by jswk** [comment]
Also, there's still sass deprecdated `/` in our code:
```
   ╷
52 │         left: calc(25% - #{$rating-point-size / 6});
   │                            ^^^^^^^^^^^^^^^^^^^^^^
   ╵
    app/javascript/stylesheets/_tours.scss 52:28      @import
    app/javascript/stylesheets/application.scss 22:9  root stylesheet

   ╷
56 │         left: calc(50% - #{$rating-point-size / 2});
   │                            ^^^^^^^^^^^^^^^^^^^^^^
   ╵
    app/javascript/stylesheets/_tours.scss 56:28      @import
    app/javascript/stylesheets/application.scss 22:9  root stylesheet

   ╷
60 │         left: calc(75% - #{$rating-point-size / 6 * 5});
   │                            ^^^^^^^^^^^^^^^^^^^^^^
   ╵
    app/javascript/stylesheets/_tours.scss 60:28      @import
    app/javascript/stylesheets/application.scss 22:9  root stylesheet

   ╷
72 │     top: #{$rating-point-size / 2 - $feedback-line-height / 2};
   │            ^^^^^^^^^^^^^^^^^^^^^^
   ╵
    app/javascript/stylesheets/_tours.scss 72:12      @import
    app/javascript/stylesheets/application.scss 22:9  root stylesheet
```

**Comment by jswk** [comment]
I would also set `config.active_support.deprecation = :raise` in `/config/environments/developement.rb` and `/config/environments/test.rb`, so that we don't accumulate deprecations. See https://www.fastruby.io/blog/rails/upgrades/deprecation-warnings-rails-guide.html#best-practices.

File: spec/features/tours/profile_spec.rb (hunk 1 above)
Comment (goreck888) [reply spec/features/tours/profile_spec.rb]: Done.

**Comment by jswk** [comment]
> PG: It would be fixed in a separate PR. (in https://github.com/cyfronet-fid/marketplace/pull/2424#issuecomment-979095607)

It's also been caused by the updates, why not correct it here too?

**Comment by jswk** [comment]
> Because it changes the way of front building, but I cherry-picked this commit here.

I applied the changes from the patch below (correcting the deprecations), and the application works without any other front-building changes (I run it with `bin/server`). I didn't have to apply other changes from the "Use faster building tools" commit.

Why not just correct the `_tours.scss` and switch to newer building tools in another PR?

```
Index: app/javascript/stylesheets/_tours.scss
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/javascript/stylesheets/_tours.scss b/app/javascript/stylesheets/_tours.scss
--- a/app/javascript/stylesheets/_tours.scss	(revision f8cf3d603271e5ab2c0fb257d0db75e091c3730e)
+++ b/app/javascript/stylesheets/_tours.scss	(date 1638365032506)
@@ -1,3 +1,5 @@
+@use "sass:math";
+
 $feedback-line-height: 7px;
 $rating-point-size: 30px;
 
@@ -49,15 +51,15 @@
       }
 
       &:nth-child(2) {
-        left: calc(25% - #{$rating-point-size / 6});
+        left: calc(25% - math.div($rating-point-size, 6));
       }
 
       &:nth-child(3) {
-        left: calc(50% - #{$rating-point-size / 2});
+        left: calc(50% - math.div($rating-point-size, 2));
       }
 
       &:nth-child(4) {
-        left: calc(75% - #{$rating-point-size / 6 * 5});
+        left: calc(75% - math.div($rating-point-size, 6) * 5);
       }
 
       &:nth-child(5) {
@@ -69,9 +71,9 @@
   .feedback-lines {
     display: inline-flex;
     position: absolute;
-    top: #{$rating-point-size / 2 - $feedback-line-height / 2};
-    left: #{$rating-point-size / 2};
-    right: #{$rating-point-size / 2};
+    top: #{math.div($rating-point-size, 2) - math.div($feedback-line-height, 2)};
+    left: math.div($rating-point-size, 2);
+    right: math.div($rating-point-size, 2);
 
     > .feedback-line {
       height: $feedback-line-height;
```

**Review by jswk — APPROVED** [review]
As the integration tests passed on @goreck888's commit in the end, I would suggest to remove the commit with e2e fixes (i.e. keep only the commits related to deprecations, e2e tests aren't affected by these changes).

**Review by kmarszalek — APPROVED** [review]

**Pull request merged**